lib/streamlit/server/server.py

```diff
@@ -663,12 +664,21 @@ def _create_app_session(self, ws: WebSocketHandler) -> AppSession:
         """
         session_data = SessionData(self._main_script_path, self._command_line)
         local_sources_watcher = LocalSourcesWatcher(session_data)
+
+        try:
+            email = ws.request.headers["X-Streamlit-User"]
```

Comment:
Could you please make the header key configurable? This would allow us to immediately leverage this feature, since we are passing the user already in a custom header.